### PR TITLE
Tsc_Refactoring _cboss

### PR DESCRIPTION
**Pull request by BINAYKUMAR943**
#### Reference Issues/PRs
#1146 TSC Refactoring for _cboss

#### What does this implement/fix? Explain your changes.

all the public methods are turned into private
input and output checks are removed from the private version
all the test cases are passing
name of capabilities dictionary is changed to _tags
new tags are added to _tag registry

#### Does your contribution introduce a new dependency? If yes, which one?

No
##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.

### CODE REVIEW

File: sktime/classification/dictionary_based/_cboss.py [hunk 1]
```diff
@@ -267,7 +264,7 @@ def fit(self, X, y):
         self._is_fitted = True
```
Comment (fkiraly) [inline sktime/classification/dictionary_based/_cboss.py:264]: this line should also be removed

File: sktime/registry/_tags.py [hunk 2]
```diff
@@ -125,11 +125,41 @@
...
+        "bool",
+        "Does classifier support limiting of max build time?",
+    ),
+   (
         "capability:pred_int",
         "forecaster",
         "bool",
         "is the forecaster capable of returning prediction intervals in predict?",
-    ),
+    )
```
Comment (fkiraly) [inline sktime/registry/_tags.py:162]: I think there should be a comma, or the linter will be unhappy

Comment (BINAYKUMAR943) [reply sktime/registry/_tags.py]: > I think there should be a comma, or the linter will be unhappy

Indeed! It was failing in the linter check due to it. I have added a comma at the end.

File: sktime/classification/dictionary_based/_cboss.py (hunk 1 above)
Comment (BINAYKUMAR943) [reply sktime/classification/dictionary_based/_cboss.py]: Thanks for reviewing. I have made this change in new commit.

> this line should also be removed

**Comment by fkiraly** [comment]
alright, all good!

Now, I'm planning to merge #1239 first (next couple hours once tests pass), which will lead to some conflicts in the tag descriptions.
You could prevent that by just copying the tags from #1239, or resolve the conflicts later.

**Comment by fkiraly** [comment]
so, it doesn't show the `_tags` file as conflicted, but I think it will lead to duplication if you leave it as-is - kindly have a look and check that each tag is contained only once.

**Comment by fkiraly** [comment]
(I could also do this, it's very quick, but I don't want to push directly to your PR in case you're also on it - let me know)

**Comment by BINAYKUMAR943** [comment]
> #1239 is merged now
> 
> Note it doesn't show the `_tags` file as conflicted, but I think it will lead to duplication if you leave it as-is - kindly have a look and check that each tag is contained only once.

thanks for the suggestion!  I have removed tags added by me as they were already added in #1239.

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**